spec/unit/recipes/site_pr_builder_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../../spec_helper'
+
+describe 'osl-jenkins::site_pr_builder' do
+  ALL_PLATFORMS.each do |p|
+    context "#{p[:platform]} #{p[:version]}" do
+      cached(:chef_run) do
+        ChefSpec::SoloRunner.new(p) do |node|
+          node.set['osl-jenkins']['credentials']['git'] = {
+            'site_pr_builder' => {
+              user: 'manatee',
+              token: 'token_password'
+            }
+          }
+          node.set['osl-jenkins']['credentials']['jenkins'] = {
+            'site_pr_builder' => {
+              user: 'manatee',
+              pass: 'password',
+              trigger_token: 'trigger_token'
+            }
+          }
+          node.set['osl-jenkins']['site_pr_builder'] = {
+            'sites_to_build' => {
+              'test-site-one' => 'https://github.com/testorg/test1.git',
+              'test-site-two' => 'https://github.com/testorg/test2.git',
+              'test-site-three' => 'https://github.com/testorg/test3.git'
+            }
+          }
+        end.converge(described_recipe)
+      end
+
+      include_context 'common_stubs'
+      it 'converges successfully' do
+        expect { chef_run }.to_not raise_error
+      end
+      it 'creates the directory' do
+        %w(test-site-one_pr_builder test-site-two_pr_builder
+           test-site-three_pr_builder).each do |job|
+          expect(chef_run).to create_directory("/var/chef/cache/#{job}").with(recursive: true)
```

Comment:
Unfortunately with ChefSpec, you need to loop through these OUTSIDE of the ``it do`` block. Also, you can remove the description in the ``it do`` block as ChefSpec automatically generates that for you. The same for the tests below.